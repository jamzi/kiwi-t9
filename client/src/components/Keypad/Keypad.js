import React from 'react';
import PropTypes from 'prop-types';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  paper: theme.mixins.gutters({
    paddingTop: 16,
    paddingBottom: 16,
    marginTop: theme.spacing.unit * 3,
    minHeight: '40px',
    display: 'flex',
    flexWrap: 'wrap',
    alignItems: 'center',
    justifyContent: 'center'
  }),
  button: {
    margin: theme.spacing.unit / 2
  },
  buttonNumber: {
    fontSize: '24px',
    color: 'darkslategray'
  },
  buttonText: {
    color: 'dimgray',
    marginLeft: '5px'
  }
});

const Keypad = props => {
  const {
    classes,
    onlyRealWords,
    handleKeyPress,
    handleShowOnlyRealWords
  } = props;

  return (
    <Paper className={classes.paper} elevation={4}>
      <div>
        <div className={classes.row}>
          <Button
            variant="raised"
            className={classes.button}
            onClick={() => handleKeyPress('1')}
          >
            <div className={classes.buttonNumber}>1</div>
            <div className={classes.buttonText} />
          </Button>
          <Button
            variant="raised"
            className={classes.button}
            onClick={() => handleKeyPress('2')}
          >
            <div className={classes.buttonNumber}>2</div>
            <div className={classes.buttonText}>abc</div>
          </Button>
          <Button
            variant="raised"
            className={classes.button}
            onClick={() => handleKeyPress('3')}
          >
            <div className={classes.buttonNumber}>3</div>
            <div className={classes.buttonText}>def</div>
          </Button>
        </div>
        <div className={classes.row}>
          <Button
            variant="raised"
            className={classes.button}
            onClick={() => handleKeyPress('4')}
          >
            <div className={classes.buttonNumber}>4</div>
            <div className={classes.buttonText}>ghi</div>
          </Button>
          <Button
            variant="raised"
            className={classes.button}
            onClick={() => handleKeyPress('5')}
          >
            <div className={classes.buttonNumber}>5</div>
            <div className={classes.buttonText}>jkl</div>
          </Button>
          <Button
            variant="raised"
            className={classes.button}
            onClick={() => handleKeyPress('6')}
          >
            <div className={classes.buttonNumber}>6</div>
            <div className={classes.buttonText}>mno</div>
          </Button>
        </div>
        <div className={classes.row}>
          <Button
            variant="raised"
            className={classes.button}
            onClick={() => handleKeyPress('7')}
          >
            <div className={classes.buttonNumber}>7</div>
            <div className={classes.buttonText}>prqs</div>
          </Button>
          <Button
            variant="raised"
            className={classes.button}
            onClick={() => handleKeyPress('8')}
          >
            <div className={classes.buttonNumber}>8</div>
            <div className={classes.buttonText}>tuv</div>
          </Button>
          <Button
            variant="raised"
            className={classes.button}
            onClick={() => handleKeyPress('9')}
          >
            <div className={classes.buttonNumber}>9</div>
            <div className={classes.buttonText}>wxyz</div>
          </Button>
        </div>
        <div className={classes.row}>
          <Button variant="raised" className={classes.button}>
            <div className={classes.buttonNumber}>*</div>
            <div className={classes.buttonText} />
          </Button>
          <Button
            variant="raised"
            className={classes.button}
            onClick={() => handleKeyPress('del')}
          >
            <div className={classes.buttonNumber}>0</div>
            <div className={classes.buttonText}>del</div>
          </Button>
          <Button variant="raised" className={classes.button}>
            <div className={classes.buttonNumber}>#</div>
            <div className={classes.buttonText} />
          </Button>
        </div>
        <FormControlLabel
          control={
            <Checkbox
              checked={onlyRealWords}
              onChange={handleShowOnlyRealWords}
              value="onlyRealWords"
            />
          }
          label="Show only real words"
        />
      </div>
    </Paper>
  );
};

Keypad.propTypes = {
  classes: PropTypes.object.isRequired,
  onlyRealWords: PropTypes.bool.isRequired,
  handleKeyPress: PropTypes.func.isRequired,
  handleShowOnlyRealWords: PropTypes.func.isRequired
};

export default withStyles(styles)(Keypad);
