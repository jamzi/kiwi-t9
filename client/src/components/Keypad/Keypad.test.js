import React from 'react';
import ReactDOM from 'react-dom';
import Keypad from './Keypad';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const handleKeyPress = jest.fn();
  const handleShowOnlyRealWords = jest.fn();

  const tree = renderer
    .create(
      <Keypad
        onlyRealWords={true}
        handleKeyPress={handleKeyPress}
        handleShowOnlyRealWords={handleShowOnlyRealWords}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
