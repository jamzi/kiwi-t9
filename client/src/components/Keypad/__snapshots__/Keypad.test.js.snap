// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root-5 MuiPaper-elevation4-11 MuiPaper-rounded-6 Keypad-paper-1"
>
  <div>
    <div
      className={undefined}
    >
      <button
        className="MuiButtonBase-root-48 MuiButton-root-32 MuiButton-raised-38 Keypad-button-2"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-33"
        >
          <div
            className="Keypad-buttonNumber-3"
          >
            1
          </div>
          <div
            className="Keypad-buttonText-4"
          />
        </span>
        <span
          className="MuiTouchRipple-root-51"
        />
      </button>
      <button
        className="MuiButtonBase-root-48 MuiButton-root-32 MuiButton-raised-38 Keypad-button-2"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-33"
        >
          <div
            className="Keypad-buttonNumber-3"
          >
            2
          </div>
          <div
            className="Keypad-buttonText-4"
          >
            abc
          </div>
        </span>
        <span
          className="MuiTouchRipple-root-51"
        />
      </button>
      <button
        className="MuiButtonBase-root-48 MuiButton-root-32 MuiButton-raised-38 Keypad-button-2"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-33"
        >
          <div
            className="Keypad-buttonNumber-3"
          >
            3
          </div>
          <div
            className="Keypad-buttonText-4"
          >
            def
          </div>
        </span>
        <span
          className="MuiTouchRipple-root-51"
        />
      </button>
    </div>
    <div
      className={undefined}
    >
      <button
        className="MuiButtonBase-root-48 MuiButton-root-32 MuiButton-raised-38 Keypad-button-2"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-33"
        >
          <div
            className="Keypad-buttonNumber-3"
          >
            4
          </div>
          <div
            className="Keypad-buttonText-4"
          >
            ghi
          </div>
        </span>
        <span
          className="MuiTouchRipple-root-51"
        />
      </button>
      <button
        className="MuiButtonBase-root-48 MuiButton-root-32 MuiButton-raised-38 Keypad-button-2"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-33"
        >
          <div
            className="Keypad-buttonNumber-3"
          >
            5
          </div>
          <div
            className="Keypad-buttonText-4"
          >
            jkl
          </div>
        </span>
        <span
          className="MuiTouchRipple-root-51"
        />
      </button>
      <button
        className="MuiButtonBase-root-48 MuiButton-root-32 MuiButton-raised-38 Keypad-button-2"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-33"
        >
          <div
            className="Keypad-buttonNumber-3"
          >
            6
          </div>
          <div
            className="Keypad-buttonText-4"
          >
            mno
          </div>
        </span>
        <span
          className="MuiTouchRipple-root-51"
        />
      </button>
    </div>
    <div
      className={undefined}
    >
      <button
        className="MuiButtonBase-root-48 MuiButton-root-32 MuiButton-raised-38 Keypad-button-2"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-33"
        >
          <div
            className="Keypad-buttonNumber-3"
          >
            7
          </div>
          <div
            className="Keypad-buttonText-4"
          >
            prqs
          </div>
        </span>
        <span
          className="MuiTouchRipple-root-51"
        />
      </button>
      <button
        className="MuiButtonBase-root-48 MuiButton-root-32 MuiButton-raised-38 Keypad-button-2"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-33"
        >
          <div
            className="Keypad-buttonNumber-3"
          >
            8
          </div>
          <div
            className="Keypad-buttonText-4"
          >
            tuv
          </div>
        </span>
        <span
          className="MuiTouchRipple-root-51"
        />
      </button>
      <button
        className="MuiButtonBase-root-48 MuiButton-root-32 MuiButton-raised-38 Keypad-button-2"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-33"
        >
          <div
            className="Keypad-buttonNumber-3"
          >
            9
          </div>
          <div
            className="Keypad-buttonText-4"
          >
            wxyz
          </div>
        </span>
        <span
          className="MuiTouchRipple-root-51"
        />
      </button>
    </div>
    <div
      className={undefined}
    >
      <button
        className="MuiButtonBase-root-48 MuiButton-root-32 MuiButton-raised-38 Keypad-button-2"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-33"
        >
          <div
            className="Keypad-buttonNumber-3"
          >
            *
          </div>
          <div
            className="Keypad-buttonText-4"
          />
        </span>
        <span
          className="MuiTouchRipple-root-51"
        />
      </button>
      <button
        className="MuiButtonBase-root-48 MuiButton-root-32 MuiButton-raised-38 Keypad-button-2"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-33"
        >
          <div
            className="Keypad-buttonNumber-3"
          >
            0
          </div>
          <div
            className="Keypad-buttonText-4"
          >
            del
          </div>
        </span>
        <span
          className="MuiTouchRipple-root-51"
        />
      </button>
      <button
        className="MuiButtonBase-root-48 MuiButton-root-32 MuiButton-raised-38 Keypad-button-2"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-33"
        >
          <div
            className="Keypad-buttonNumber-3"
          >
            #
          </div>
          <div
            className="Keypad-buttonText-4"
          />
        </span>
        <span
          className="MuiTouchRipple-root-51"
        />
      </button>
    </div>
    <label
      className="MuiFormControlLabel-root-58"
    >
      <span
        className="MuiButtonBase-root-48 MuiIconButton-root-70 MuiSwitchBase-root-66 MuiCheckbox-root-61 MuiCheckbox-colorSecondary-65 MuiSwitchBase-checked-67 MuiCheckbox-checked-62"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={undefined}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label-75"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-76"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
          <input
            checked={true}
            className="MuiSwitchBase-input-69"
            disabled={undefined}
            id={undefined}
            name={undefined}
            onChange={[Function]}
            tabIndex={undefined}
            type="checkbox"
            value="onlyRealWords"
          />
        </span>
        <span
          className="MuiTouchRipple-root-51"
        />
      </span>
      <span
        className="MuiTypography-root-82 MuiTypography-body1-91 MuiFormControlLabel-label-60"
      >
        Show only real words
      </span>
    </label>
  </div>
</div>
`;
